### ISSUE TITLE
[5.9] Switch expression won't compile when assigning value to already defined variable.

### ISSUE THREAD

**Issue by kr1sc** [apple/swift#66412]
**Description**

Function `foo1`will not compile but function `foo2` will. Compilator error is `Cannot find 'val' in scope`. The difference is only with declaration and initialization of `bar` variable.

```
func foo1() {
    let value: Int? = nil
    
    var bar: Int = 1
    bar = switch value {
    case let val?: val
    default: 0
    }
}

func foo2() {
    let value: Int? = nil
    
    var bar: Int = switch value {
    case let val?: val
    default: 0
    }
}
```

**Steps to reproduce**

Try to compile above code in Xcode 15.0 beta.

**Expected behavior**

Both functions compile.

**Environment**
- Swift compiler version info
```
swift-driver version: 1.82.2 Apple Swift version 5.9 (swiftlang-5.9.0.114.6 clang-1500.0.27.1)
Target: arm64-apple-macosx13.0
```
- Xcode version info 
```
Version 15.0 beta (15A5160n)
```
- Deployment target: 
```
iOS 17.0
```

**Comment by hamishknight** [comment]
This is a known issue, and is noted in the Xcode 15 release notes:

> An if or switch expression that introduces a new variable may produce a spurious “cannot find ‘’ in scope” error when that if or switch is used in an assignment. (109192116)

Was recently fixed by #65893

**Comment by hamishknight** [comment]
This should be fixed in Xcode 15 beta 3
